rust/opendp-ffi/src/core.rs

```diff
@@ -38,6 +35,79 @@ impl FfiObject {
     }
 }
 
+#[repr(C)]
+pub struct FfiError {
+    pub variant: *mut c_char,
+    pub message: *mut c_char, // MAY BE NULL!
+    pub backtrace: *mut c_char,
+}
+
+impl FfiError {
+    pub fn new(error: ErrorBase) -> *mut Self {
+        fn substring(s: &str, start: usize, end: usize) -> String {
+            assert!(end >= start);
+            let len = end - start;
+            s.chars().skip(start).take(len).collect()
+        }
+        let error_string = format!("{:?}", error.error);
+        let type_message = error_string.find("(").map_or_else(
+            || (error_string.clone(), None),
+            |i| (substring(&error_string, 0, i), Some(substring(&error_string, i + 2, error_string.len() - 2)))
+        );
+        let backtrace = format!("{:?}", error.backtrace);
+        let ffi_error = FfiError {
+            variant: util::into_c_char_p(type_message.0),
+            message: type_message.1.map_or(ptr::null::<c_char>() as *mut c_char, util::into_c_char_p),
+            backtrace: util::into_c_char_p(backtrace)
+        };
+        util::into_raw(ffi_error)
```

Comment:
I did some reworking of the error type in `32-error-handling-glue-python-macros` to clean this up. Up to you if it goes in this PR.

Reply:
Specifically, here: https://github.com/opendp/opendp/commit/651219dee278be6394f26936319b7bf438ef46d0#diff-126584c448e8cd1780d21068c8134f1e3dd2f4a89e73d93f466f11d4e9a96f41L46-R52

Reply:
I think this is a nice improvement, I've merged the changes.